opba-banking-protocol-facade/src/test/java/de/adorsys/opba/protocol/facade/services/ais/ServiceContextProviderTest.java

```diff
@@ -3,102 +3,151 @@
                         FacadeServiceableRequest.builder()
-                                .bankId(testBankID)
-                                .serviceSessionId(session.getId())
+                                .serviceSessionId(sessionId)
                                 .fintechRedirectUrlOk("http://google.com")
                                 .fintechRedirectUrlNok("http://microsoft.com")
+                                .authorizationSessionId(uriFacadeResult.getAuthorizationSessionId())
+                                .redirectCode(uriFacadeResult.getRedirectCode())
                                 .build()
                 ).build();
         ServiceContext<FacadeServiceableGetter> providedContext2 = serviceContextProvider.provide(request2);
+        EncryptionService encryptionService2 = providedContext2.getEncryptionService();
+
+        transactionTemplate.execute(t -> secondRequestCheck(sessionId, encryptionService2));
+    }
 
-        assertThat(providedContext2.getRequest().getFacadeServiceable().getBankId()).isEqualTo(testBankID);
+    @SneakyThrows
+    private UUID checkSavedSession(EncryptionService encryptionService, FacadeServiceableRequest facadeServiceable) {
+
+        ServiceSession session = serviceSessionRepository.findAll().iterator().next();
+
+        // check that key is recoverable with password
+        KeyWithParamsDto keyWithParams = secretKeyOperations.generateKey(
+                PASSWORD,
+                session.getAlgo(),
+                session.getSalt(),
+                session.getIterCount()
+        );
+        assertThat(secretKeyOperations.decrypt(session.getSecretKey())).isEqualTo(keyWithParams.getKey());
+
+        // check that in context stored first request parameters facadServicable
+        String context = session.getContext();
+        byte[] decryptedContext = encryptionService.decrypt(context.getBytes());
+        FacadeServiceableRequest decryptedFacadeServiceble =
+                MAPPER.readValue(decryptedContext, FacadeServiceableRequest.class);
+        assertThat(decryptedFacadeServiceble).isEqualToComparingFieldByField(facadeServiceable);
+
+        // storing some data to context using provided encryption service
+        String encryptedContext = new String(encryptionService.encrypt(
+                MAPPER.writeValueAsBytes(PROTOCOL_DEFINED_DATA_TO_STORE_IN_CONTEXT))
+        );
+        session.setContext(encryptedContext);
+        session.setProtocol(protocolRepository.findById(3L).orElseThrow(
+                () -> new IllegalArgumentException("protocol 3 not found")));
+        serviceSessionRepository.save(session);
+        return session.getId();
     }
 
-    @EnableXs2aProtocol
-    @EnableBankingPersistence
-    @SpringBootApplication(scanBasePackages = "de.adorsys.opba.protocol.facade")
-    public static class TestConfig {
+    @SneakyThrows
+    private Void secondRequestCheck(UUID sessionId, EncryptionService encryptionService2) {
```

Comment:
Either make this test simpler or create separate test that checks only one thing - columns with key and context in database, after session is opened are encrypted